client/src/components/Home.js

```diff
@@ -117,7 +117,31 @@ const Home = ({ user, logout }) => {
     [setConversations]
   );
 
+  const updateReadTimestamps = useCallback(
+    (...usernames) => {
+      const conversationIds = []
+      setConversations((prev) => 
+        prev.map((convo) => {
+          if (usernames.includes(convo.otherUser.username)) {
+            const convoCopy = { ...convo };
+            conversationIds.push(convoCopy.id);
+            convoCopy.lastSeen = new Date();
+            return convoCopy;
+          } else {
+            return convo;
+          }
+        })
+      )
+
+      axios.patch("/api/conversations/read", {
+        conversationIds
+      });
```

Comment:
There's one thing I'm not quite understanding about this: Why did you choose to use an array, instead of sending just an id (since it's always just going to be one user or none that matches the if condition; `usernames.includes(convo.otherUser.username)` )
Also, as a suggestion, you could return from the function before the API request if there is no id in the array:

```suggestion
      const conversationIds = []
      setConversations((prev) => 
        prev.map((convo) => {
          if (usernames.includes(convo.otherUser.username)) {
            const convoCopy = { ...convo };
            conversationIds.push(convoCopy.id);
            convoCopy.lastSeen = new Date();
            return convoCopy;
          } else {
            return convo;
          }
        })
      )
      
      if (!conversationIds.length) return;

      axios.patch("/api/conversations/read", {
        conversationIds
      });
```

Reply:
Oh it's because I send both the id for the conversation someone just closed and the conversation they just opened. I wanted to send it in one message rather than two separate messages. My reasoning being that if someone closes a conversation the lastSaw data for them should reflect that.